app/code/Magento/Cms/Block/Adminhtml/Wysiwyg/Images/Content/Uploader.php

```diff
@@ -71,4 +71,18 @@ protected function _getMediaType()
         }
         return $this->getRequest()->getParam('type');
     }
+
+    /**
+     * Return allowed file extensions based on request or config
+     *
+     * @return array
+     */
+    protected function _getAllowedFileExtensions()
+    {
+        if ($encodedExtensions = $this->getRequest()->getParam('allowed_extensions')) {
+            // phpcs:ignore Magento2.Functions.DiscouragedFunction
+            return(explode(" ", base64_decode($encodedExtensions)));
```

Comment:
@piotrekkaminski This does not look safe for me. Could you take a look?

Reply:
What would you recommend. I couldn't think of a better way of passing the list of file types into the media gallery component than this. Open to suggestions.